app/validators/questionnaire_schema.py

```diff
@@ -536,3 +536,16 @@ def get_answers_from_question(question):
             *question.get("dynamic_answers", {}).get("answers", []),
             *question.get("answers", []),
         ]
+
+    def get_section_block_ids(self, current_section=None):
+        block_ids = []
+        section = self.get_section(current_section)
```

Comment:
Will this work if current section is None? is there a reason to call it current section instead of just section?

I think you can simplify this function to just return
```
self.blocks_by_section_id[section]
```